contracts/modules/LoanOpenings/LoanOpenings.sol

```diff
@@ -70,160 +57,89 @@ contract LoanOpenings is State, LoanOpeningsEvents, VaultController, InterestHan
         }
 
         // get required collateral
-        uint256 collateralAmountRequired = _getRequiredCollateral(
-            loanParamsLocal.loanToken,
-            loanParamsLocal.collateralToken,
-            sentValues[1],
-            initialMargin,
-            isTorqueLoan
-        );
+        uint256 collateralAmountRequired = _getRequiredCollateral(loanParamsLocal.loanToken, loanParamsLocal.collateralToken, sentValues[1], initialMargin, isTorqueLoan);
         require(collateralAmountRequired != 0, "collateral is 0");
 
-        return _borrowOrTrade(
-            loanParamsLocal,
-            loanId,
-            isTorqueLoan,
-            collateralAmountRequired,
-            initialMargin,
-            sentAddresses,
-            sentValues,
-            loanDataBytes
-        );
+        return _borrowOrTrade(loanParamsLocal, loanId, isTorqueLoan, collateralAmountRequired, initialMargin, sentAddresses, sentValues, loanDataBytes);
+    }
+
+    function getDefaultLoanParamsAndStore(
+        address loanToken,
+        address collateralToken,
+        bool isTorqueLoan
+    ) internal returns (LoanParams memory loanParamsLocal) {
+        bool isDefault;
+        (loanParamsLocal, isDefault) = getDefaultLoanParams(loanToken, collateralToken, isTorqueLoan);
+        if (isDefault) {
+            // store if its default
+            loanParams[loanParamsLocal.id] = loanParamsLocal;
+        }
+    }
+    // collateralToken is passed from the iToken and there it guarantees no invalid value can be passed
+    function getDefaultLoanParams(
```

Comment:
throw an error if it is a non torque loan and id is equal to 0. There is no check for that here so margin trades are able to pass through

Reply:
its a view function. can't throw an error. empty result is the error which is checked afterwards

Reply:
We decided to keep margin trade, but it did not return anything empty which was the problem. We can keep it as is since we are allowing margin trading any pair